source/uwp/Renderer/lib/ActionHelpers.cpp

```diff
@@ -870,18 +871,30 @@ namespace AdaptiveCards::Rendering::Uwp::ActionHelpers
 
             if (currentButtonIndex < maxActions && mode == winrt::AdaptiveCards::ObjectModel::Uwp::ActionMode::Primary)
             {
-                // If we have fewer than the maximum number of actions and this action's mode is primary, make a button
-                actionControl = CreateActionButtonInActionSet(adaptiveCard,
-                                                              adaptiveActionSet,
-                                                              action,
-                                                              currentButtonIndex,
-                                                              actionsPanel,
-                                                              showCardsStackPanel,
-                                                              columnDefinitions,
-                                                              renderContext,
-                                                              renderArgs);
-
-                currentButtonIndex++;
+                try
+                {
+                    // If we have fewer than the maximum number of actions and this action's mode is primary, make a button
+                    actionControl = CreateActionButtonInActionSet(adaptiveCard,
+                                                                  adaptiveActionSet,
+                                                                  action,
+                                                                  currentButtonIndex,
+                                                                  actionsPanel,
+                                                                  showCardsStackPanel,
+                                                                  columnDefinitions,
+                                                                  renderContext,
+                                                                  renderArgs);
+
+                    currentButtonIndex++;
+                }
+                catch (winrt::hresult_error const& ex)
+                {
+                    // We want to continue if the error is E_FAIL
```

Comment:
even if E_FAIL is returned only when fallback mechanism is missing. this can change in the future, and could we be better off with a specific error message for missing fallback?

Reply:
@jwoo-msft I just updated this - let me know what you think